src/server/req_runjob.c

```diff
@@ -989,6 +989,17 @@ svr_strtjob2(job *pjob, struct batch_request *preq)
 	job_attr_def[(int) JOB_ATR_Comment].at_decode(
 			&pjob->ji_wattr[(int) JOB_ATR_Comment], (char *) 0,
 			(char *) 0, "");
+	if (pjob->ji_qs.ji_svrflags & JOB_SVFLG_SubJob &&
+			0 == subjob_index_to_offset(pjob->ji_parentaj,
+					get_index_from_jid(pjob->ji_qs.ji_jobid))) {
```

Comment:
I'd suggest putting the == 0 at the end.  I suggested this style when the PBS coding standard was being written and it was rejected.  In any case, the reason to use 0 == X is so you don't accidently do X = 0.  In this case, X is a function, so the compiler would complain anyway.